### ISSUE TITLE
[Regression] Updating Emsdk from 3.1.31 to 3.1.32 causes undefined symbol error when building OpenSSL

### ISSUE THREAD

**Issue by Ravbug** [emscripten-core/emsdk#1195]
On 3.1.32, I get the following linker error:
`wasm-ld: error: RavEngine/deps/GameNetworkingSockets/openssl-cmake/crypto/libcrypto.a(dso_dlfcn.c.o): undefined symbol: dladdr`
This error does not occur on 3.1.31. 

The file in question: [dso_dlfcn.c](https://github.com/openssl/openssl/blob/master/crypto/dso/dso_dlfcn.c)

**Build environment information:**
OS: Ubuntu 22.04 for x86_64 on GitHub Actions
Compiler: GNU 11.3.0 using CMake toolchain file `emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake`
Other compile flags:
- definitions: 
   - `-sUSE_SDL=2`
   - `-pthread`
   - `-fexceptions`
- Link arguments: 
   - `-s MAX_WEBGL_VERSION=2`
   - `s MIN_WEBGL_VERSION=2`
   - `-s FULL_ES3=1`
   - `-s USE_WEBGPU`
   - `-s GL_ASSERTIONS=1`
   - `-s OFFSCREEN_FRAMEBUFFER=1`
   - `-s OFFSCREENCANVAS_SUPPORT=1`
   - `-s GL_DEBUG=1`
   - `-fexceptions`
   - `-s PROXY_TO_PTHREAD`
   - `-s USE_PTHREADS`
   - `-s LLD_REPORT_UNDEFINED`
   - `-s NO_DISABLE_EXCEPTION_CATCHING`
   - `-s NO_DISABLE_EXCEPTION_THROWING`
   - `-s PTHREAD_POOL_SIZE=4`
   - `-s ASSERTIONS=1`
   - `-s ALLOW_MEMORY_GROWTH=1`
   - `-s MAXIMUM_MEMORY=4GB`
   - `-s USE_SDL=2`
   - `-s WASM_BIGINT`
   - `-s TOTAL_MEMORY=150MB`

**Comment by sbc100** [comment]
Fix is in https://github.com/emscripten-core/emscripten/pull/18896.   Although a better solution (at least for the codesize of your application) is probably to teach the build that dladdr (and indeed all of the dlfnc function) are not available.

emscripten includes stub functions for them, but it will trap if you try to call them since (without `MAIN_MODULE` set at least) these functions are not implemented.   

In other words, if your program does not actually depend on dynamic linking it should probably not included references to those symbols.

**Comment by Ravbug** [comment]
I disabled dlfnc for my emscripten build and it works, thank you!
